### ISSUE TITLE
Host starting while player is on stats screen causes them to be stuck on In Game screen.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [chrsmith/hedgewars#645]
```
What steps will reproduce the problem?
1.  Stay on game stats screen on end of a game
2.  Room admin starts a new game.
3.  On end of game, you will be stuck on the In Game screen.

This is a result of allowing admins to begin games without everyone being ready.
Since dag10 made the change in frontend, I guess he is technically owner of 
this.
Is he coming back?

This is obv safe to fix in the .19 branch since it does not desync.

Leaving PAGE_INGAME, entering PAGE_GAMESTATS
Client:  ("ROUNDFINISHED", "0") 
Server:  ("CLIENT_FLAGS", "-g", "nemo") 
Server:  ("EM", "Ayv6mA==") 
Server:  ("EM", "Ayv+gQ==") 
Leaving PAGE_GAMESTATS, entering PAGE_GAMESTATS


```

Original issue reported on code.google.com by `kyberneticist@gmail.com` on 1 Jun 2013 at 4:02

**Comment by GoogleCodeExporter** [comment]
Original comment by `kyberneticist@gmail.com` on 1 Jun 2013 at 4:03

**Comment by GoogleCodeExporter** [comment]
```
Most likely caused by bug 432, not by dag10, removing the owner tag...
CCing unC0Rr as it could be a simple server change involved.
```

Original comment by `vittorio...@gmail.com` on 1 Jun 2013 at 8:22

**Comment by GoogleCodeExporter** [comment]
```
Assigned due to:
changeset:   8384:e691acd778f5
user:        dag10
date:        Mon Jan 21 02:21:04 2013 -0500
summary:     It is now possible to start a game even if not all players are 
"ready"; a confirmation prompt is shown. This commit updates the server.

IMO this would be better fixed in the frontend in page swapping, thus the 
reason I linked.

But, eh, guess I'll let dag10 and unc0rr sort that out.

```

Original comment by `kyberneticist@gmail.com` on 1 Jun 2013 at 9:38

**Comment by GoogleCodeExporter** [comment]
```
Can't be fixed server-side, since server has no idea you're still on stats 
page. Purely frontend's issue.
```

Original comment by `unC0Rr` on 2 Jun 2013 at 12:38

**Comment by GoogleCodeExporter** [comment]
```
I that's what I thought :)
So I figured dag10 would be a bit more familiar w/ that section of code having 
added the initial frontend portion...

My usual reflex is when a commit introduces a bug, to assign initial owner to 
committer.

```

Original comment by `kyberneticist@gmail.com` on 2 Jun 2013 at 3:12

**Comment by GoogleCodeExporter** [comment]
```
Issue 650 has been merged into this issue.
```

Original comment by `kyberneticist@gmail.com` on 2 Jun 2013 at 3:13

**Comment by GoogleCodeExporter** [comment]
Original comment by `kyberneticist@gmail.com` on 2 Jun 2013 at 3:18
* Changed title: **Host starting while player is on stats screen causes them to be stuck on In Game screen.**

**Comment by GoogleCodeExporter** [comment]
```
Well, reverting to old behavior on the server would prevent the bug.
```

Original comment by `unC0Rr` on 2 Jun 2013 at 6:15

**Comment by GoogleCodeExporter** [comment]
```
Then we'll get bugs about the confirm prompt not working :-p

Although admittedly not as bad as getting stuck on a screen that forces a 
restart...
We should maybe test on m8y in case that exposes some new frontend confusion of 
state :)

```

Original comment by `kyberneticist@gmail.com` on 3 Jun 2013 at 12:03

**Comment by GoogleCodeExporter** [comment]
```
whoops, I did a mistake, not finding e691acd778f5
readding dag10 in CC
```

Original comment by `vittorio...@gmail.com` on 3 Jun 2013 at 7:17

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 3f5e0280db4d (not tested though)
```

Original comment by `unC0Rr` on 3 Jun 2013 at 8:16
* Changed state: **Fixed**
